DataProcessing/datax-host/src/main/scala/datax/input/KafkaStreamingFactory.scala

```diff
@@ -0,0 +1,94 @@
+// *********************************************************************
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License
+// *********************************************************************
+package datax.input
+
+import java.sql.Timestamp
+
+import datax.constants.ProductConstant
+import datax.telemetry.AppInsightLogger
+import datax.utility.DateTimeUtil
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.apache.log4j.LogManager
+import org.apache.spark.rdd.RDD
+import org.apache.spark.streaming.{StreamingContext, Time}
+import org.apache.spark.streaming.kafka010.ConsumerStrategies.Subscribe
+import org.apache.spark.streaming.kafka010.KafkaUtils
+import org.apache.spark.streaming.kafka010.LocationStrategies.PreferConsistent
+
+// Factory object that generates the DStream from Kafka input and invokes processor
+object KafkaStreamingFactory extends  StreamingFactory[ConsumerRecord[String,String]]{
+
+  private def getKafkaConf(kafkaInput:InputKafkaConf) = {
+    val logger = LogManager.getLogger("KafkaConfBuilder")
+
+    val groupId = kafkaInput.groupId
+    logger.warn("kafka group.id=" + groupId)
+
+   var kafkaParams = Map[String, Object](
+      "bootstrap.servers" -> kafkaInput.bootStrapServers,
+      "key.deserializer" -> classOf[StringDeserializer],
+      "value.deserializer" -> classOf[StringDeserializer],
+      "group.id" -> groupId,
+      "auto.offset.reset" -> kafkaInput.autoOffsetReset,
+      "heartbeat.interval.ms" -> kafkaInput.heartBeatInterval,
+      "session.timeout.ms" -> kafkaInput.sessionTimeout,
+      "fetch.max.wait.ms" -> kafkaInput.fetchMaxWait
+   )
+
+    // EventHub for Kafka needs few additional params
+    if(kafkaInput.isEventHubKafka) {
+      kafkaParams += (
+        "security.protocol" -> kafkaInput.securityProtocol,
+        "sasl.mechanism" -> kafkaInput.saslMechanism,
+        "sasl.jaas.config" -> kafkaInput.saslConfig
+      )
+    }
+
+    //TODO: enable external store based checkpoint
+
+    kafkaParams
+  }
+
+  def getStream(streamingContext: StreamingContext,
+                inputConf:InputConf,
```

Comment:
instead of passing in a `InputConf`, would be better to pass in the `SettingDictionary`.

Reply:
and the factory parse the flat settings into its known structure of configuration.

Reply:
Strongly typed config is a better experience. This way, the parsing of settings dictionary is handled in the corresponding InputSettings class.